lib/block-patterns.php

```diff
@@ -236,6 +236,58 @@ function load_remote_patterns() {
 	}
 }
 
+/**
+ * Register featured patterns from wordpress.org/patterns.
+ *
+ * @since 5.9.0
+ */
+function load_remote_featured_patterns() {
+	/**
+	 * Filter to disable remote block patterns.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool $should_load_remote
+	 */
+	$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+
+	if ( ! $should_load_remote ) {
+		return;
+	}
+
+	// TODO: check about the `featured` name.. Should this be reserved?
+	// Also check core logic: https://developer.wordpress.org/reference/classes/wp_block_pattern_categories_registry/
+	// Do we just allow collisions/overide with categories when they have the same name? Needs checking..
+	if ( ! WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'Featured' ) ) {
+		register_block_pattern_category( 'featured', array( 'label' => __( 'Featured', 'gutenberg' ) ) );
+	}
+
+	$request = new WP_REST_Request( 'GET', '/wp/v2/pattern-directory/patterns' );
+	// TODO: change this with new category id when created in patterns directory..
+	$request->set_param( 'category', 4 ); // temp gallery category id.
```

Comment:
I would prefer if we could have a custom endpoint for featured patterns. But this isn't a blocker to ship the feature.

Reply:
Why do you think we should use a custom endpoint/wrapper for this? We can get the needed results by setting a `category` or `keyword` (this will be decided in pattern directory though).. Is it because the hardcoded `4`?

Reply:
I agree this will probably be decided in the pattern directory.

> Is it because the hardcoded 4?

Yeah, I would prefer not to include implementation details like this in code. But like I mentioned this isn't a blocker 👍

Reply:
It seems like this makes sense as a category, since you want it to work like one. The `keyword` taxonomy is likely going to be deprecated/internal-only (in favor of a meta field).

Reply:
It's not important for me to be a category, I just want to have a way to fetch them 😄 .

>Yeah, we decided not to launch with it because the technical details were unclear. If it's just another pattern category, we can create that any time.

Having a category though makes sense to me too.